producer/src/block/block_test.go

```diff
@@ -165,4 +178,3 @@ func TestDeserialize(t *testing.T) {
 		t.Errorf("Blocks do not match")
```

Comment:
This test is failing.

Reply:
I believe this test is failing is because the dataLen field in the expected block is initialized to 0 and not set to the length of Data.
If I include: expected.DataLen = uint16(len(expected.Data)) in the test function, this test will pass.

Reply:
Ah I overlooked something. We will need to update the `Deserialize` test. Initialize `DataLen` similarly to how the `TestSerialize` does it.

Reply:
In TestSerialize, DataLen is set after the test block is initialized. Should I initialize DataLen along with the other members during the initialization of the test block as well?

Reply:
In Go, if you omit initializing a field it just zeros it out. So you can omit initializing `DataLen` when the other fields are initialized and just initialize it after like 
`expected.DataLen = uint16(len(expected.Data))`